src/discovery.jl

```diff
@@ -295,83 +290,81 @@ function find_host_compiler(toolkit_version=nothing)
             gcc_ver = VersionNumber(m.captures[1])
             @trace("Found GCC $gcc_ver at $gcc_path")
 
-            if toolkit_version == nothing || in(gcc_ver, gcc_range)
+            if toolkit_version == nothing || gcc_supported(gcc_ver, toolkit_version)
                 push!(gcc_possibilities, (gcc_path, gcc_ver))
             end
         end
 
         # select the most recent compiler
         if length(gcc_possibilities) == 0
-            if gcc_range == nothing
-                error("Could not find any GCC")
-            else
-                error("Could not find a suitable GCC (your CUDA v$toolkit_version needs GCC $gcc_range).")
-            end
+            error("Could not find a suitable GCC")
         end
         sort!(gcc_possibilities; rev=true, lt=(a, b) -> a[2]<b[2])
         host_compiler, host_version = gcc_possibilities[1]
     elseif Compat.Sys.iswindows()
         # Windows: search for MSVC in default locations
-        try
-            global msvc_paths
-            msvc_paths = String[]
-            arch = Sys.WORD_SIZE == 64 ? "amd64" : "x86"
-
-            # locate VS2017
-            vswhere = joinpath(get(ENV, "ProgramFiles(x86)", ""), "Microsoft Visual Studio", "Installer", "vswhere.exe")
-            if isfile(vswhere)
-                msvc_cmd_tools_dir = chomp(read(`$vswhere -latest -property installationPath`, String))
-                vs_prompt = joinpath(msvc_cmd_tools_dir, "VC", "Auxiliary", "Build", "vcvarsall.bat")
-                tmpfile = tempname()
-                run(pipeline(`$vs_prompt $arch \& where cl.exe`, tmpfile))
-                msvc_path = readlines(tmpfile)[end]
-                push!(msvc_paths, msvc_path)
-            end
 
-            # locate VS2012 to 2014
-            vc_versions_100_140 = ["VS140COMNTOOLS", "VS120COMNTOOLS", "VS110COMNTOOLS", "VS100COMNTOOLS"]
-            for inds in find(x -> haskey(ENV, x), vc_versions_100_140)
-                path = ENV[vc_versions_100_140[inds]]
-                msvc_path = joinpath(dirname(dirname(dirname(path))), "VC", "bin", arch, "cl.exe")
-                push!(msvc_paths, msvc_path)
-            end
-        catch ex
-            error("Compatible Visual Studio installation cannot be found; Visual Studio 2017, 2015, 2013, 2012, or 2010 is required.")
+        # discover Visual Studio installations
+        msvc_paths = String[]
+        arch = Sys.WORD_SIZE == 64 ? "amd64" : "x86"
+        ## locate VS2017
+        vswhere = joinpath(get(ENV, "ProgramFiles(x86)", ""), "Microsoft Visual Studio", "Installer", "vswhere.exe")
```

Comment:
On 32 bit windows this should  just be `joinpath(get(ENV, "ProgramFiles", "")` although I can't test.